### PR TITLE
Configure log for VxWorks

### PR DESCRIPTION
**Pull request by kuhlenough**
IPC and Default factories use APIs VX does not have

### CODE REVIEW

**Comment by Lastique** [comment]
What are the missing APIs? I'm especially interested about the default factories.

**Comment by kuhlenough** [comment]
I'm investigating, and will reopen soon with an improved request

**Comment by kuhlenough** [comment]
Here is a better pull with excellent test results. Still a couple IPC tests throwing odd exceptions, but they look related to other libraries other than log.

**Comment by Lastique** [comment]
The object name scopes are intended to be non-overlapping. For example, the same object name in the global and process group scopes should identify different objects. This changes with this PR.

If there isn't any replacement to the POSIX functions on VxWorks, I think we should use the closest reasonable replacement, but with a distinct prefix. For instance, we could assume that all processes running under the same user belong to the same session and process group. Thus for `object_name::process_group` the prefix should be "pgid." + `getuid()` and for `object_name::session` it should be "sid." + `getuid()`. If the necessary API appears in the future we can replace `getuid` with it rather easily.

**Comment by kuhlenough** [comment]
That sounds like a reasonable approach.  I'll update the request.

File: src/dump.cpp [hunk 1]
```diff
@@ -192,7 +192,8 @@ struct function_pointer_initializer
     static void cpuid(uint32_t& eax, uint32_t& ebx, uint32_t& ecx, uint32_t& edx)
     {
 #if defined(__GNUC__)
-#if defined(__i386__) && (defined(__PIC__) || defined(__PIE__)) && !(defined(__clang__) || (defined(BOOST_GCC) && BOOST_GCC >= 50100))
+#if defined(__i386__) && (defined(__PIC__) || defined(__PIE__)) && !(defined(__clang__) || (defined(BOOST_GCC) && BOOST_GCC >= 50100)) \
+	||( defined(__VXWORKS__) && defined(__PIC__))
```
[the remark this answers is not in the document]
Comment (kuhlenough) [reply src/dump.cpp]: I'll double check, but I think it was failure in GCC 4.x 64bit IA compile that threw an undefined opcode on the other branch.

File: example/wide_char/main.cpp [hunk 2]
```diff
@@ -77,12 +81,14 @@ void init_logging()
             << "> " << expr::message
     );
 
+#if !defined(BOOST_LOG_NO_LOCALE)
```
[the remark this answers is not in the document]
Comment (kuhlenough) [reply example/wide_char/main.cpp]: VxWorks locale support is pretty basic, timezone support, not much more.  Supporting boost.locale is long way off. So why don't I just skip building the example for Vxworks in the Jamfile instead? That way we don't stall the test suite.

[the remark this answers is not in the document]
Comment (Lastique) [reply example/wide_char/main.cpp]: Boost.Locale can use other libraries like iconv or icu to implement some of its functionality. But disabling the example in the Jamfile is fine with me.

File: src/dump.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lastique) [reply src/dump.cpp]: Yes, and a comment that VxWorks reserves ebx even in 64-bit mode would be useful. Thanks.

[the remark this answers is not in the document]
Comment (kuhlenough) [reply src/dump.cpp]: Done, please see the update pull request.

**Comment by Lastique** [comment]
Merged to develop, thanks. Sorry for the delay.

**Pull request merged**